hlapipeline/alignimages.py

```diff
@@ -175,127 +176,190 @@ def perform_align(input_list, archive=False, clobber=False, update_hdr_wcs=False
+                tweakwcs.tweak_image_wcs(imglist, reference_catalog, match=match)
+                # Interpret RMS values from tweakwcs
+                interpret_fit_rms(imglist, reference_catalog)
+
+                # determine the quality of the fit
+                rms_fit, num_fit  = determine_fit_quality(imglist, print_fit_parameters=print_fit_parameters)
+
+                # update the best fit 
+                if rms_fit < best_fit:
+                   best_fit = rms_fit
+                   best_num_fit = num_fit
+                   for item in imglist:
+                       item.best_meta = item.meta.copy()
+
+
 
+    # 7: Write new fit solution to input image headers
+    print("-------------------- STEP 7: Update image headers with new WCS information --------------------")
+    if best_fit < MAX_FIT_RMS:
+       print("The fitting process was successful with a best fit total rms of {} mas".format(best_fit))
+    else:
+       print("The fitting process was unsuccessful with a best fit total rms of {} mas".format(best_fit))
+
+    if best_fit < MAX_FIT_LIMIT:
+        # update to the meta information with the lowest rms if it is reasonable
+        for img in imglist:
+            item.meta = item.best_meta
+
+
+    if update_hdr_wcs:
+
+        update_image_wcs_info(imglist, processList)
+        print("\nSUCCESS")
+    else:
+        print("\n STEP SKIPPED")
+    return (0)
+
+
+def determine_fit_quality(imglist, print_fit_parameters=True):
+    """Determine the quality of the fit to the data
+
+    Parameters
+    ----------
+    imglist: list
+        output of interpret_fits. Contains sourcelist tables, newly computed WCS info, etc. for every chip of every valid
+        input image.  This list should have been  updated, in-place, with the new RMS values;
+        specifically,
+
+            * 'FIT_RMS': RMS of the separations between fitted image positions and reference positions
+            * 'TOTAL_RMS': mean of the FIT_RMS values for all observations
+            * 'NUM_FITS': number of images/group_id's with successful fits included in the TOTAL_RMS
+
+        These entries are added to the 'tweakwcs_info' dictionary.
+
+    Returns
+    -------
+    max_rms_val : float
+        The best Totol rms dteremined from all of the images
+
+    num_xmatches: int
+        The number of stars used in matching the data
+
+    """
+    tweakwcs_info_keys = OrderedDict(imglist[0].meta['tweakwcs_info']).keys()
+    imgctr=0
+    max_rms_val = 1e9
+    num_xmatches = 0
+    for item in imglist:
+        image_name = item.meta['name']
+        #Handle fitting failures (no matches found)
+        if item.meta['tweakwcs_info']['status'].startswith("FAILED") == True:
+                print("No cross matches found in any catalog for {} - no processing done.".format(image_name))
+                continue
+        fit_rms_val = item.meta['tweakwcs_info']['FIT_RMS']
+        max_rms_val = item.meta['tweakwcs_info']['TOTAL_RMS']
```

Comment:
Setting 'max_rms_val' at this point only reports the RMS value for each group_id(image), which we already determined did not represent the correct RMS for the VISIT.